docs: improve guide for installing from source

albertony opened the pull request:
#### What is the purpose of this change?

Main purpose is to updated according to Go version 1.17: Use `go install` instead of `go get`.

Included additional information regarding cmount buildtag, required winfsp installation on windows, and "optimized builds" with `-ldflags -s`.

Maybe the text should be split into subsections now? E.g. the "git + go build" method vs the "go install/go get" method? (Not sure what to name the headers then)

#### Was the change discussed in an issue or in the forum before?

Fixes #5809

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

ncw reviewed (commented):
Sorry seem to have lost this PR!

Can you rebase it and update it (go1.18 is now current and go1.16 is the minimum) and I will merge - thank you, and sorry for the delay.

ncw commented:
> Sure. And also; I think it should get some updates due to your recent cgofuse/cmount changes. E.g. the following, about building with "-tags cmount":
> 
> > This assumes you have a GCC compatible C compiler (GCC or Clang) in your PATH. On Windows you can use the MinGW port of GCC,
> 
> and:
> 
> > and set environment variable CPATH pointing to the fuse include directory within the WinFsp installation.
> 
> Should probably be updated to mention that on Windows one can instead build with `SET CGO_ENABLED=0`, and then avoid GCC, MinGW, CPATH etc, and this is how the official release version is built - starting with the upcoming 1.59 release(?)

Yes that is definitely worth mentioning.

I should probably make `cmount` build by default on Windows without the `-tags cmount` just like normal `cmd/mount` does. That would hurt my brain with the build constraints though!

albertony commented:
Rebased and updated with information about Windows build with cmount but without cgo. Also added a note about `-trimpath`, resulting in the following as the "close to official" build command:
```
go build -trimpath -ldflags -s -tags cmount
```

ncw reviewed (approved):
LGTM :-) I'll merge now.
